scripts/scripts.js

```diff
@@ -714,7 +714,76 @@ async function loadLazy(doc) {
     setTimeout(() => {
       const element = hash ? main.querySelector(hash) : false;
       if (element) {
-        element.scrollIntoView();
+        // Handle lazy-loaded images that might cause content shift
+        const handleLazyLoadedImages = () => {
+          // Find all images above the target element that are lazy-loaded
+          const allElements = Array.from(main.querySelectorAll('*'));
+          const targetIndex = allElements.indexOf(element);
+
+          if (targetIndex === -1) {
+            element.scrollIntoView();
+            return;
+          }
+
+          // Get all lazy-loaded images before the target element
+          const precedingImages = allElements
+            .slice(0, targetIndex)
+            .filter((el) => {
+              if (el.tagName === 'IMG' && el.getAttribute('loading') === 'lazy') {
+                return true;
+              }
+              // Also check for images inside picture elements
+              if (el.tagName === 'PICTURE') {
+                const img = el.querySelector('img[loading="lazy"]');
+                return img !== null;
+              }
+              return false;
+            })
+            .map((el) => (el.tagName === 'PICTURE' ? el.querySelector('img') : el));
+
+          if (precedingImages.length === 0) {
+            // No lazy-loaded images above target, scroll immediately
+            element.scrollIntoView();
+            return;
+          }
+
+          // Track loaded images
+          let loadedCount = 0;
+          const totalImages = precedingImages.length;
+
+          // Function to scroll after each image loads
+          const scrollAfterLoad = () => {
+            loadedCount += 1;
+            // Always scroll to the element to account for content shift
+            element.scrollIntoView({ behavior: 'smooth', block: 'start' });
+
+            // Remove listeners from remaining images if all are loaded
+            if (loadedCount === totalImages) {
+              precedingImages.forEach((img) => {
+                img.removeEventListener('load', scrollAfterLoad);
+                img.removeEventListener('error', scrollAfterLoad);
+              });
+            }
+          };
+
+          // Add load listeners to all preceding lazy images
+          precedingImages.forEach((img) => {
+            if (img.complete) {
+              // Image already loaded
+              loadedCount += 1;
```

Comment:
Don't we need to add a scroll here just in case?

Reply:
no, if it's loaded, it won't expand anymore and push lower elements down.